### PR TITLE
Release 6.1.0 -- Filter

### PR DESCRIPTION
**Pull request by Schparky**
### Added
- New `Filter` config option on RestAPI adapter

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,6 +6,127 @@ interface. The runtime for this application is configured using a `config.json`
...
+```json
+{
+  "Source": {
+    "Type": "RestAPI",
+    "ExtraJSON": {
+      "Pagination": {
+        "Scheme": "pages",
+        "NumberKey": "page",
+        "PageSizeKey": "page-size",
+        "FirstPage": "1",
```
Comment (Baggerone) [inline README.md:79]: Sorry. I hadn't finished reviewing the other PR and got distracted.
This should be `FirstIndex`, right?

File: README.md [hunk 2]
```diff
@@ -6,6 +6,127 @@ interface. The runtime for this application is configured using a `config.json`
...
+data. If the value of an attribute configured in a filter is empty or null, the
+record is not included in the output data.
+
+#### Properties
+- Attribute -- The name of the attribute to filter on. Does not need to be listed in the sync attributes.
+- Expression -- A text expression for which to search. Uses RE2 regular expression syntax.
+
+#### Example config
+
+Following is an example configuration for Pagination. Unrelated parameters have
```
Comment (Baggerone) [inline README.md:107]: Instead of `Pagination`, `Filtering`?

File: internal/person.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+type Person struct {
+	CompareValue   string
+	ID             string
+	Attributes     map[string]string
+	DisableChanges bool
+}
+
+func (p *Person) Matches(filters Filters) bool {
+	for _, f := range filters {
+		value := p.Attributes[f.Attribute]
```
Comment (Baggerone) [inline internal/person.go:12]: What if the f.Attribute doesn't match any of the p.Attributes **keys**?

Should this be something like
```
value, ok := p.Attributes[f.Attribute]
if !ok {
   continue
}
...
```

**Review by Baggerone — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (Schparky) [reply README.md:79]: Correct, thanks.

File: README.md (hunk 2 above)
Comment (Schparky) [reply README.md:107]: Thanks

File: internal/person.go (hunk 3 above)
Comment (Schparky) [reply internal/person.go:12]: Not necessarily. I wrote the documentation to cover this. If the attribute value is missing or empty, the record is omitted. So `false` is the correct return value in this case. Are you thinking this treatment is problematic?

Comment (Baggerone) [reply internal/person.go:12]: What if f.Attribute has a typo? Won't that mean that no records will be included?

Comment (Schparky) [reply internal/person.go]: Yes, I will cause the sync to fail if the attribute is missing.

**Review by Baggerone — APPROVED** [review]

**Pull request merged**